dice-where-downloader-lib/src/main/java/technology/dice/dicewhere/downloader/destination/local/LocalFileAcceptor.java

```diff
@@ -39,6 +40,10 @@ public StreamConsumer<Void> getStreamConsumer(
         LOG.debug("Destination directory already exists");
       }
       Files.copy(stream, destination, StandardCopyOption.REPLACE_EXISTING);
+      if ((!noMd5Check) && (!originalFileMd5.matches(stream.md5()))) {
```

Comment:
Can we do this check at the top, so we avoid the copy+delete?

Reply:
No, because you only know the md5 of the transfer, after the transfer is complete